accepted/2020/mono-convergence/platform-specific-httpclient.md

```diff
@@ -0,0 +1,456 @@
+* Platform settings. Some platform-level settings aren't exposed in a way that SocketsHttpHandler could view them but are directly integrated with the native HTTP stack.
+
+### Platform-specific Http properties need to be available for customization
+In cases where HTTP handler wraps platform-specific implementation and the implementation has a way to control HTTP connection behaviour or properties, such properties need to be exposed to C# developers on the public platform-specific type.  If HttpClientHandler itself exposes such a knob, it needs to control the associated customization on the underlying platform.
+
+A few examples of platform handlers control properties
+
+* [cachePolicy](https://developer.apple.com/documentation/foundation/nsurlrequest/cachepolicy/useprotocolcachepolicy)
+* [allowsConstrainedNetworkAccess](https://developer.apple.com/documentation/foundation/nsmutableurlrequest/3325676-allowsconstrainednetworkaccess)
+* [allowsExpensiveNetworkAccess](https://developer.apple.com/documentation/foundation/nsmutableurlrequest/3325677-allowsexpensivenetworkaccess)
+* [networkServiceType](https://developer.apple.com/documentation/foundation/nsmutableurlrequest/1412378-networkservicetype)
+* [ReadTimeout](https://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#setReadTimeout(int))
+* [AllowUserInteraction](https://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#setAllowUserInteraction(boolean))
+
+## Xamarin.iOS/tvOS/watchOS
+There are 3 handlers available (SocketsHttpHandler, CFNetworkHandler, and NSUrlSessionHandler), with **NSUrlSessionHandler** as the default (except on watchOS, where the only supported handler is NSUrlSessionHandler).
+
+The handler is only preselected as the default with [UI options available](https://docs.microsoft.com/en-us/xamarin/cross-platform/macios/http-stack#selecting-an-httpclient-stack) using VS Project Settings dialog to change them. This is required to “fix” existing HttpClient code which was written with the assumption that the default handler will be the one working best on the platform.
+
+The ILLinker is used to remove the unnecessary handlers from the final binary in this case.
+
+### Handlers in Detail
+**SocketsHttpHandler Handler**
+
+It's a managed implementation, but for a few very specific things, mainly related to getting proxy information from the OS, it does P/Invoke. The handler behaves generally the same across all platforms where sockets are available. The downside of using it is mostly felt in networking connection switching when needed, for examaple when switching back and forth between the wifi/cellular antennas.
+
+It's a managed implementation, but for a few very specific things, mainly related to getting proxy information from the OS, we do P/Invoke.
```

Comment:
The sentence `It's a managed implementation, but for a few very specific things, mainly related to getting proxy information from the OS, we do P/Invoke.` is duplicated.

Reply:
Fixed